src/topology/basic.lean

```diff
@@ -491,26 +511,71 @@ end
 @[simp] lemma nhds_ne_bot {a : α} : 𝓝 a ≠ ⊥ :=
 ne_bot_of_le_ne_bot pure_ne_bot (pure_le_nhds a)
 
-lemma interior_eq_nhds {s : set α} : interior s = {a | 𝓝 a ≤ principal s} :=
+/-!
+### Cluster points
+-/
+
+/-- A point `x` is a cluster point of a filter `F` if 𝓝 x ⊓ F ≠ ⊥. -/
+def cluster_pt (x : α) (F : filter α) : Prop := 𝓝 x ⊓ F ≠ ⊥
+
+lemma cluster_pt.of_le_nhds {x : α} {f : filter α} (H : f ≤ 𝓝 x) (h : f ≠ ⊥) : cluster_pt x f :=
+by rwa [cluster_pt, inf_comm, inf_eq_left.mpr H]
+
+lemma cluster_pt.of_nhds_le {x : α} {f : filter α} (H : 𝓝 x ≤ f) : cluster_pt x f :=
+by simp [cluster_pt, inf_eq_left.mpr H]
+
+lemma cluster_pt.mono {x : α} {f g : filter α} (H : cluster_pt x f) (h : f ≤ g) :
+  cluster_pt x g :=
+ne_bot_of_le_ne_bot H $ inf_le_inf_left _ h
+
+lemma cluster_pt_of_inf_left {x : α} {f g : filter α} (H : cluster_pt x $ f ⊓ g) :
+  cluster_pt x f :=
+H.mono inf_le_left
+
+lemma cluster_pt_of_inf_right {x : α} {f g : filter α} (H : cluster_pt x $ f ⊓ g) :
+  cluster_pt x g :=
+H.mono inf_le_right
+
+/-- A point `x` is a cluster point of a sequence `u` if it is a cluster point
+of `map u at_top`. -/
+def seq_cluster_pt {ι :Type*} [preorder ι] (x : α) (u : ι → α) : Prop := cluster_pt x (map u at_top)
```

Comment:
I'm not sure that we should hardcode `at_top`. It makes sense to talk, e.g., about cluster points of `f x` as `x → a`.

Also it would be nice to have a lemma `cluster_pt x (map u l) ↔ ∀ s ∈ 𝓝 x, ∃^f a in l, u a ∈ s`.